Recent commit breaks adruino framework install by using c++17 structure where arduino only supports 11

In [this patch](https://github.com/echavet/MitsubishiCN105ESPHome/commit/0e5e805b7b7cb56640373b44abd7437f6dcca2f7) the [line here](https://github.com/echavet/MitsubishiCN105ESPHome/commit/0e5e805b7b7cb56640373b44abd7437f6dcca2f7#diff-d43a38446bde4439d9f5e2cf9281a9641f0b2e5051498eb731967b104d3a7d29R163-R167) is added but that is only valid in esf-idf framework and not the arduino framework. Installing presents the following compile error:

```
Compiling .pioenvs/esphome-web-046268/src/esphome/core/application.cpp.o
src/esphome/components/cn105/climateControls.cpp: In lambda function:
src/esphome/components/cn105/climateControls.cpp:163:20: warning: structured bindings only available with -std=c++17 or -std=gnu++17
         for (auto& [k, v] : v) {
                    ^
Compiling .pioenvs/esphome-web-046268/src/esphome/core/color.cpp.o
src/esphome/components/cn105/climateControls.cpp: In member function 'void esphome::CN105Climate::controlTemperature()':
src/esphome/components/cn105/climateControls.cpp:190:49: error: 'clamp' is not a member of 'std'
         this->wantedSettings.temperature = std::clamp(setting, 10.0f, 31.0f);
                                                 ^~~~~
src/esphome/components/cn105/climateControls.cpp:190:49: note: suggested alternative:
In file included from src/esphome/components/api/proto.h:5,
                 from src/esphome/components/api/api_pb2.h:5,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/climateControls.cpp:1:
src/esphome/core/helpers.h:104:41: note:   'esphome::clamp'
 template<typename T> constexpr const T &clamp(const T &v, const T &lo, const T &hi) {
                                         ^~~~~
Compiling .pioenvs/esphome-web-046268/src/esphome/core/component.cpp.o
Compiling .pioenvs/esphome-web-046268/src/esphome/core/component_iterator.cpp.o
Compiling .pioenvs/esphome-web-046268/src/esphome/core/controller.cpp.o
Compiling .pioenvs/esphome-web-046268/src/esphome/core/entity_base.cpp.o
*** [.pioenvs/esphome-web-046268/src/esphome/components/cn105/climateControls.cpp.o] Error 1
Compiling .pioenvs/esphome-web-046268/src/esphome/core/helpers.cpp.o
========================== [FAILED] Took 4.54 seconds ==========================
```

--- a/components/cn105/climateControls.cpp
+++ b/components/cn105/climateControls.cpp
@@ -160,8 +160,8 @@ static float mapCelsiusForConversionFromFahrenheit(const float c) {
             {81, 27.0}, {82, 27.5}, {83, 28.0}, {84, 28.5}, {85, 29.0},
             {86, 29.5}, {87, 30.0}, {88, 30.5}
         };
-        for (auto& [k, v] : v) {
-            k = (k - 32.0f) / 1.8f;
+        for (auto& pair : v) {
+            pair.first = (pair.first - 32.0f) / 1.8f;
         }
         return *new std::map<float, float>(v.begin(), v.end());
     }();
@@ -187,7 +187,7 @@ void CN105Climate::controlTemperature() {
         this->wantedSettings.temperature = this->lookupByteMapIndex(TEMP_MAP, 16, (int)(setting + 0.5)) > -1 ? setting : TEMP_MAP[0];
     } else {
         setting = std::round(2.0f * setting) / 2.0f;  // Round to the nearest half-degree.
-        this->wantedSettings.temperature = std::clamp(setting, 10.0f, 31.0f);
+        this->wantedSettings.temperature =  setting < 10 ? 10 : (setting > 31 ? 31 : setting);
     }
 }
 

--- a/components/cn105/hp_readings.cpp
+++ b/components/cn105/hp_readings.cpp
@@ -192,8 +192,8 @@ static float mapCelsiusForConversionToFahrenheit(const float c) {
             {27.0, 81}, {27.5, 82}, {28.0, 83}, {28.5, 84}, {29.0, 85},
             {29.5, 86}, {30.0, 87}, {30.5, 88}
         };
-        for (auto& [k, v] : *m) {
-            v = (v - 32.0f) / 1.8f;
+        for (auto& pair : *m) {
+            pair.second = (pair.second - 32.0f) / 1.8f
         }
         return *m;
     }();

--- a/components/cn105/hp_readings.cpp
+++ b/components/cn105/hp_readings.cpp
@@ -193,7 +193,7 @@ static float mapCelsiusForConversionToFahrenheit(const float c) {
             {29.5, 86}, {30.0, 87}, {30.5, 88}
         };
         for (auto& pair : *m) {
-            pair.second = (pair.second - 32.0f) / 1.8f
+            pair.second = (pair.second - 32.0f) / 1.8f;
         }
         return *m;
     }();

